src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.cs

```diff
@@ -222,6 +222,16 @@ private Instrumenter Instrumenter
 
         private BoundExpression? VisitExpressionImpl(BoundExpression node)
         {
+            if (node is BoundNameOfOperator nameofOperator)
+            {
+                Debug.Assert(!nameofOperator.WasCompilerGenerated);
+                var nameofIdentiferSyntax = (IdentifierNameSyntax)((InvocationExpressionSyntax)nameofOperator.Syntax).Expression;
```

Comment:
>(IdentifierNameSyntax) [](http://example.com/codeflow?start=44&length=22)

Minor: It looks like the cast may be unnecessary.

Reply:
I think so, but if this isn't an identifier, I'd like to fail fast here.